pykrige/ok.py

```diff
@@ -743,10 +758,16 @@ def execute(self, style, xpoints, ypoints, mask=None, backend='vectorized',
                 raise ValueError('Specified backend {} for a moving window '
                                  'is not supported.'.format(backend))
         else:
-            bd = cdist(xy_points,  xy_data, 'euclidean')
-            if self.coordinates_type == 'geographic':
+            if self.coordinates_type == 'euclidean':
+                bd = cdist(xy_points,  xy_data, 'euclidean')
+            elif self.coordinates_type == 'geographic':
                 # Convert euclidean distances to great circle distances:
-                bd = core.euclid3_to_great_circle(bd)
+                # Note: xy_points_c & xy_data_c are in a packed-complex format
+                # Note: Using packed-complex because cdist won't work

```

Comment:
Could you please elaborate on this? cdist does work in ND space with N=2 that should be similar to complex, or am I missing something?

Reply:
I did not see where cdist handles 'geographic' lat-lon distances. It looked like you'd have to pass a lat-lon distance function in. Using Numpy array multiplication syntax seemed an efficient workaround. The bigger issue was that in some places Euclidean distances are used where geographic distances should be used for consistency.  In fact, since this pull request, I found another place where Euclidean distances are incorrectly used regardless of the geographic flag. There's nothing wrong with core.euclid3_to_great_circle, per se.

Reply:
Also at this point, it may be better to reject this pull request, since I've another [branch](https://github.com/michaelleerilee/PyKrige/tree/fit_custom_variogram_too) which has gone beyond this, treating the issue in a better way, reflecting a couple of months more study of the code. FWIW, ok.py's _get_krige_matrix also needs revision consistent with the way geographic coordinates are treated. I've not worked on the problem everywhere, just the flow that goes through the OK vectorized backend, which I needed for my work.

Reply:
This part of the latlon-coordinate handling should be okay in the original code.
In line [706](https://github.com/bsmurphy/PyKrige/pull/99/files/9f438d1ed556f7b547c2b2b02871ccabb0822b2e#diff-51079543933ff8bb6612fa4380cefbe3R706), the lon-lat-coordinates have been converted to 3d Euclidean coordinates, hence cdist + conversion.
This was, at the time, the least invasive change I could think of. Surely direct application of great circle distance should be better in the long run. I think the use of the KDTree for n_closest_points may have influenced my decision. If there's a way to use that with spherical coordinates, core.euclid3_to_great_circle could (and should probably) be omitted completely.